### ISSUE TITLE
Ability to authenticate SDK using trusted profile tokens

### ISSUE THREAD

**Issue by kavya498** [IBM/keyprotect-go-client#70]
As of now SDK uses only API key way of authentication..

The ask is, if user gives trusted_profile_id, SDK should be able to generate tokens and then use that token for authentication
Other SDKs that are used in terraform provider supports this feature..

ref: https://github.com/IBM/go-sdk-core/blob/main/v5/core/vpc_instance_authenticator.go

**Comment by mrodden** [comment]
Yep, trusted profiles are a rather new feature of IAM which I have not had the time to dig into and understand how it would fit into the current IAM token API that the code exposes as its public API today.
